### DISCUSSION TITLE
Mesh object for creating internal surface?

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#28033] [Q&A Meshing] [answered]
Hi, I'm trying to create an internal surface in a rectangular mesh to use an interface BC with. This is the setup:

![box2](https://github.com/idaholab/moose/assets/52802063/d8959eeb-61d4-4175-8089-3a35ea8927ad)

(NB, the surface does not cut through the blue box). Previously, with the case where theta = 0, I had simply created two different blocks in the lower (black) rectangle with BoundingBoxNodeSetGenerator (even though both sides had the same material properties), and then I used SideSetsBetweenSubdomainsGenerator to define the surface.

Now that I have an angle, is it still possible to do this, or would I need to try to do it directly from the mesh generation software? I was having issues with that in the past (refining specific subregions within the meshing software and still retaining domain names) hence why I did the subdomain definitions within MOOSE.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Can you use the ParsedGenerateSideset?
this can work with an equation, if the nodes exist already at the right location

Guillaume

**Reply by richmondodufisan** [reply]
this should work, thanks. Is there a way to run an input file and have it output the mesh (without running the whole simulation) with the moose-defined sidesets viewable in paraview? Just to be sure that my equation is defining it appropriately.

**Reply by GiudGiud** [reply]
add --mesh-only on the command line after the input file name

**Reply by richmondodufisan** [reply]
thanks. but I just realized that actually, this might not work- I'm defining a variable jump so I need to be able to define different variables on each side of the sideset. Would there be a way to split the block into two based on the defined sideset?

**Reply by GiudGiud** [reply]
if the expression you use to define the sideset is good enough for ParsedGenerateSideset, then it will work very similarly in ParsedSubdomainGenerator which you can use to split the mesh into blocks

**Reply by richmondodufisan** [reply]
hmm okay this should work, thank you!
